docs/terms/evaluation_parameter.md

```diff
@@ -1,15 +1,152 @@
+
+```python title="Python code"
+eval_param_urn = 'urn:great_expectations:validations:my_expectation_suite_1:expect_table_row_count_to_be_between.result.observed_value'
+downstream_batch.expect_table_row_count_to_equal(
+    value={
+        '$PARAMETER': eval_param_urn, # this is the actual parameter we're going to use in the validation
+    }
+)
+```
+
+The core of this is a `$PARAMETER : URN` pair. When Great Expectations encounters a `$PARAMETER` flag during validation, it will replace the `URN` with a value retrieved from an Evaluation Parameter Store or Metrics Store.
+
+If you do not have a previous Expectation Suite's Validation Results to reference, however, you can instead provide Evaluation Parameters with a temporary initial value. For example, the interactive method of creating Expectations is based on Validating Expectations against a previous run of the same Expectation Suite.  Since a previous run has not been performed when Expectations are being created, Evaluation Parameters cannot reference a past Validation and will require a temporary value instead.  This will allow you to test Expectations that are meant to rely on values from previous Validation runs before you have actually used them to Validate data.  
+
+Say you have created a Pandas dataframe called `my_df` that references a CSV that contains ten rows of data.  (To do so in the interactive process for creating Expectations, you would use the `great_expectations.read_csv()` method).  You want to create an expression that asserts that the row count for each Validation remains the same as the previous `upstream_row_count`, but since there is no previous `upstream_row_count` you need to provide a value that matches what the Expectation you are creating will find.
+
+To do so, you would define the parameter in question (`upstream_row_count`) by assigning it to the `$PARAMETER` value in a dictionary.  Then, you would provide the temporary value for that parameter by setting it as the value of the `$PARAMETER.<parameter_in_question>` key in the same dictionary.  Or, in this case, the `$PARAMETER.upstream_row_count`.
+
+For an example of this, see below:
+
+```python title="Python code"
+import great_expectations
+csv_path = 'path/to/my_data.csv'
+my_df = great_expectations.read_csv(csv_path)
+my_df.expect_table_row_count_to_equal(
+    value={"$PARAMETER": "upstream_row_count", "$PARAMETER.upstream_row_count": 10},
+    result_format={'result_format': 'BOOLEAN_ONLY'}
+)
+```
+
+This will return `{'success': True}`.
+
+An alternative method of defining the temporary value for an Evaluation Parameter is the `set_evaluation_parameter()` method, as shown below:
+
+```python title="Python code"
+my_df.set_evaluation_parameter("upstream_row_count", 10)
+```
+
+If the Evaluation Parameter's value is set in this way, you do not need to set it again (or define it alongside the use of the `$PARAMETER` key) for future Expectations.
```

Comment:
We need to update these examples to not reference "my_df" and instead reference a validator-based flow. (Those examples come from the v2 api)

Reply:
I've updated them to use a validator, and to explain that the validator can be gotten as part of the interactive process for creating Expectations.  I also tested all of the updated examples (one was broken in both V2 and V3!)